src/realm-settings/key-providers/ecdsa-generated/ECDSAGeneratedForm.tsx

```diff
@@ -0,0 +1,360 @@
+          render={({ onChange, value }) => {
+            return (
+              <Switch
+                id="kc-active"
+                label={t("common:on")}
+                labelOff={t("common:off")}
+                isChecked={value[0] === "true"}
+                data-testid={
+                  value[0] === "true"
+                    ? "internationalization-enabled"
+                    : "internationalization-disabled"
+                }
+                onChange={(value) => {
+                  onChange([value.toString()]);
+                }}
+              />
+            );
+          }}
+        />
+      </FormGroup>
+      <FormGroup
+        label={t("ellipticCurve")}
+        fieldId="kc-elliptic-curve"
+        labelIcon={
+          <HelpItem
+            helpText="realm-settings-help:ellipticCurve"
+            forLabel={t("ellipticCurve")}
+            forID="kc-elliptic-curve"
+          />
+        }
+      >
+        <Controller
+          name="config.ecdsaEllipticCurveKey"
+          control={form.control}
+          defaultValue={["P-256"]}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-ecdsa-elliptic-curve"
+              onToggle={() => setIsKeySizeDropdownOpen(!isKeySizeDropdownOpen)}
+              onSelect={(_, value) => {
+                onChange([value.toString()]);
+                setIsKeySizeDropdownOpen(false);
+              }}
+              selections={[value.toString()]}
+              isOpen={isKeySizeDropdownOpen}
+              variant={SelectVariant.single}
+              aria-label={t("ellipticCurve")}
+              data-testid="select-elliptic-curve-size"
+            >
+              {ecdsaEllipticCurveOptions?.map((item) => (
+                <SelectOption
+                  selected={item === value}
+                  key={item}
+                  value={item}
+                />
+              ))}
+            </Select>
+          )}
+        />
+      </FormGroup>
+      <ActionGroup className="kc-ecdsa-form-buttons">
+        <Button
+          className="kc-ecdsa-form-save-button"
+          data-testid="add-provider-button"
+          variant="primary"
+          type="submit"
+        >
+          {t("common:save")}
+        </Button>
+        <Button
+          className="kc-ecdsa-form-cancel-button"
+          onClick={(!editMode && handleModalToggle) || undefined}
+          variant="link"
+        >
+          {t("common:cancel")}
+        </Button>
+      </ActionGroup>
+    </FormAccess>
+  );
+};
+
+export const ECDSAGeneratedSettings = () => {
+  const { t } = useTranslation("realm-settings");
+  const providerId = useRouteMatch<MatchParams>(
+    "/:realm/realm-settings/keys/:id?/:providerType?/settings"
+  )?.params.providerType;
+  return (
+    <>
+      <ViewHeader titleKey={t("editProvider")} subKey={providerId} />
+      <PageSection variant="light">
+        <ECDSAGeneratedForm providerType={providerId!} editMode />
```

Comment:
Using the non-null assertion operator can lead to exceptions at run-time.
```suggestion
        <ECDSAGeneratedForm providerType={providerId} editMode />
```

Reply:
A type error is yielded here unless the non-null assertion operator is used

Reply:
If that is the case then the props of the `ECDSAGeneratedForm` component will have to be changed so the value can be `undefined`.